evalml/pipelines/components/transformers/imputers/imputer.py

```diff
@@ -67,6 +67,10 @@ def fit(self, X, y=None):
 
         self._all_null_cols = set(X.columns) - set(X.dropna(axis=1, how='all').columns)
         X_copy = X.copy()
+
+        if (X.dtypes == bool).all():
+            X_copy = X_copy.astype('category')
```

Comment:
@angela97lin heads up these changes will cause conflicts for #1423 (update all components to accept woodwork datatables). I think they should be pretty straightforward to address--when you're resolving the conflicts you could call `dt.set_logical_types(...)` to do this transformation, or you could keep doing the transformation with pandas. Great that @christopherbunn is adding test coverage for this case, that'll help!

Reply:
Ah, #1423 was just merged!

@christopherbunn if you need a hand updating this, please ping @angela97lin or I

Reply:
Oo yup, just saw this. Yes yes, please reach out if you need help fixing the merge conflicts!